src/components/Stadium/Input.tsx

```diff
@@ -1,22 +1,17 @@
 import TextField, {TextFieldProps} from "@mui/material/TextField";
-import {ChangeEvent} from "react";
+import React from "react";
 
 interface InputProps {
     type: "text" | "number";
     placeholder?: string;
     id?: string;
     name?: string;
     inputRef?: React.Ref<HTMLInputElement>;
-    value: string;
-    onChange: (e: ChangeEvent<HTMLInputElement>) => void;
 }
-
-const Input: React.FC<InputProps & TextFieldProps> = ({
-    onChange,
-    inputRef,
-    ...props
-}) => {
-    return <TextField {...props} inputRef={inputRef} onChange={onChange} />;
-};
+const Input = React.forwardRef<HTMLInputElement, InputProps & TextFieldProps>(
+    ({...props}, ref) => {
+        return <TextField {...props} inputRef={ref} />;
```

Comment:
Here and in select component I think you don`t need ref any more as you pass register as props
Feel free to ask if it`s not clear